odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1953,56 +1925,12 @@ def is_node_branded(self, node):
             for attr in node.attrib
         )
 
-    @tools.ormcache('self.id')
-    def get_view_xmlid(self):
-        domain = [('model', '=', 'ir.ui.view'), ('res_id', '=', self.id)]
-        xmlid = self.env['ir.model.data'].sudo().search_read(domain, ['module', 'name'])[0]
-        return '%s.%s' % (xmlid['module'], xmlid['name'])
-
     @api.model
     def render_public_asset(self, template, values=None):
-        template = self.sudo().browse(self.get_view_id(template))
-        template._check_view_access()
-        return template.sudo()._render(values, engine="ir.qweb")
-
-    def _render_template(self, template, values=None, engine='ir.qweb'):
-        return self.browse(self.get_view_id(template))._render(values, engine)
-
-    def _render(self, values=None, engine='ir.qweb', minimal_qcontext=False, options=None):
-        assert isinstance(self.id, int)
-
-        qcontext = dict() if minimal_qcontext else self._prepare_qcontext()
-        qcontext.update(values or {})
+        return self.env['ir.qweb']._render(template, values)
```

Comment:
Security alert: this does the same thing as the method below but without the `_`